application/console/controllers/CronController.php

```diff
@@ -82,4 +83,27 @@ public function actionGoogleAnalytics()
 
         \Yii::warning($gaEvents);
     }
+
+
+    public function actionSendDelayedMfaRelatedEmails()
+    {
+        /* @var $emailer Emailer */
+        $emailer = \Yii::$app->emailer;
+
+        $users = User::findAll();
+
+        foreach ($users as $user) {
```

Comment:
Given that this could be processing a list of thousands of users, there may be a way to get the list of Users as some kind of iterator, which only retrieves them as you're ready for them (or perhaps in batches, I'm not sure). While that may slow the process down, it would avoid having to instantiate a User model for every user in the database all at once, which I suspect will be the more likely limiting factor (RAM vs. CPU).

Reply:
Switched to use $query->each()